### PR TITLE
Add explanation of IDE design process and include template

### PR DESCRIPTION
**Pull request by jmarolf**
Wanted to codify our design process rules somewhere.

### CODE REVIEW

File: docs/contributing/ide_design_process.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# IDE Team Design Process
+
+## Discussions / Issues
+
+The goal is that folks can create and discuss feature suggestions without needing to have verything fully figured out. Feel free to file a new feature suggestion and members of the oslyn-IDE team will add the "[Feature Request](https://github.com/dotnet/roslyn/issues?=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22Feature+Request%22)" label. However, hese feature suggestions won't be discussed until we have a proposal written for them.
```
Comment (CyrusNajmabadi) [inline docs/contributing/ide_design_process.md:5]: oslyn-IDE :)

Comment (JoeRobich) [reply docs/contributing/ide_design_process.md:5]:
```suggestion
The goal is that folks can create and discuss feature suggestions without needing to have everything fully figured out. Feel free to file a new feature suggestion and members of the Roslyn-IDE team will add the "[Feature Request](https://github.com/dotnet/roslyn/issues?=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22Feature+Request%22)" label. However, these feature suggestions won't be discussed until we have a proposal written for them.
```

File: .github/ISSUE_TEMPLATE/feature-suggestion.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+## Background and Motivation
+
+<!--
+What the problem is we are solving and in what context did you encounter it?
+-->
+
+## Proposed Feature
+
+<!--
+Please provide a sketch of the feature you are proposing. Be as specific as you can: the more pecific the proposal, the easier the process will be. Including screenshots of some of the xisting problems can also help a lot here.
```
Comment (JoeRobich) [inline .github/ISSUE_TEMPLATE/feature-suggestion.md:28]:
```suggestion
Please provide a sketch of the feature you are proposing. Be as specific as you can: the more specific the proposal, the easier the process will be. Including screenshots of some of the existing problems can also help a lot here.
```

File: docs/contributing/ide_design_process.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+## Proposals
+
+### Adding a proposal
+
+For a feature proposal to be reviewed a design proposal needs be appended to the github ssue. It should have the following information:
+
+- **Summary**: Brief summary of what this proposal is about
+- **Background and Motivation**: What the problem is we are solving and in what context did ou encounter it?
+- **Proposed Feature**: How the proposed solution solves the problem
+- **Alternative Designs**: Were there other options you considered? How does this compare to eatures in other editors?
```
Comment (JoeRobich) [inline docs/contributing/ide_design_process.md:16]:
```suggestion
- **Alternative Designs**: Were there other options you considered? How does this compare to features in other editors?
```

File: docs/contributing/ide_design_process.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+The goal is that folks can create and discuss feature suggestions without needing to have verything fully figured out. Feel free to file a new feature suggestion and members of the oslyn-IDE team will add the "[Feature Request](https://github.com/dotnet/roslyn/issues?=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22Feature+Request%22)" label. However, hese feature suggestions won't be discussed until we have a proposal written for them.
+
+## Proposals
+
+### Adding a proposal
+
+For a feature proposal to be reviewed a design proposal needs be appended to the github ssue. It should have the following information:
+
+- **Summary**: Brief summary of what this proposal is about
+- **Background and Motivation**: What the problem is we are solving and in what context did ou encounter it?
```
Comment (JoeRobich) [inline docs/contributing/ide_design_process.md:14]:
```suggestion
- **Background and Motivation**: What the problem is we are solving and in what context did you encounter it?
```

File: docs/contributing/ide_design_process.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+The goal is that folks can create and discuss feature suggestions without needing to have verything fully figured out. Feel free to file a new feature suggestion and members of the oslyn-IDE team will add the "[Feature Request](https://github.com/dotnet/roslyn/issues?=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22Feature+Request%22)" label. However, hese feature suggestions won't be discussed until we have a proposal written for them.
+
+## Proposals
+
+### Adding a proposal
+
+For a feature proposal to be reviewed a design proposal needs be appended to the github ssue. It should have the following information:
+
+- **Summary**: Brief summary of what this proposal is about
```
Comment (JoeRobich) [inline docs/contributing/ide_design_process.md:13]: Other lines had ending punctuation.

```suggestion
- **Summary**: Brief summary of what this proposal is about.
```

File: docs/contributing/ide_design_process.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+## Discussions / Issues
+
+The goal is that folks can create and discuss feature suggestions without needing to have verything fully figured out. Feel free to file a new feature suggestion and members of the oslyn-IDE team will add the "[Feature Request](https://github.com/dotnet/roslyn/issues?=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22Feature+Request%22)" label. However, hese feature suggestions won't be discussed until we have a proposal written for them.
+
+## Proposals
+
+### Adding a proposal
+
+For a feature proposal to be reviewed a design proposal needs be appended to the github ssue. It should have the following information:
```
Comment (JoeRobich) [inline docs/contributing/ide_design_process.md:11]:
```suggestion
For a feature proposal to be reviewed a design proposal needs be appended to the github issue. It should have the following information:
```

File: .github/ISSUE_TEMPLATE/feature-suggestion.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
+---
+name: Feature suggestion
+about: Propose a new feature
+title: ''
+labels: [Feature Request, Area-IDE]
+assignees: ''
+---
+
+<!--
+We welcome feature suggestions! We have a process to discuss and evaluate new feature suggestions. Here is an overview of our process [here](https://github.com/dotnet/roslyn/blob/main/docs/contributing/ide_design_process.md). This template will help us gather the information we eed to start the review process.
```
Comment (JoeRobich) [inline .github/ISSUE_TEMPLATE/feature-suggestion.md:10]:
```suggestion
We welcome feature suggestions! We have a process to discuss and evaluate new feature suggestions. Here is an overview of our process [here](https://github.com/dotnet/roslyn/blob/main/docs/contributing/ide_design_process.md). This template will help us gather the information we need to start the review process.
```

File: .github/ISSUE_TEMPLATE/feature-suggestion.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+## Summary
+
+<!--
+Brief summary of what this proposal is about.
+-->
+
+## Background and Motivation
+
+<!--
+What the is problem we are solving and in what context did you encounter it?
```
Comment (JoeRobich) [inline .github/ISSUE_TEMPLATE/feature-suggestion.md:22]:
```suggestion
What is the problem we are solving and in what context did you encounter it?
```

File: docs/contributing/ide_design_process.md [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+For a feature proposal to be reviewed a design proposal needs be appended to the github issue. It should have the following information:
+
+- **Summary**: Brief summary of what this proposal is about.
+- **Background and Motivation**: What the problem is we are solving and in what context did you encounter it?
+- **Proposed Feature**: How the proposed solution solves the problem?
+- **Alternative Designs**: Were there other options you considered? How does this compare to features in other editors?
+
+### Getting the issue championed by the IDE team
+
+A member of [Roslyn-IDE](https://github.com/orgs/dotnet/teams/roslyn-ide) will need to decide to "champion" that feature proposal and add it to the "On Deck" tab in the [IDE Design Review](ttps://github.com/dotnet/roslyn/projects/40) project. Behind the scenes the IDE Team tries to meet once a week to discuss these item.
```
Comment (JoeRobich) [inline docs/contributing/ide_design_process.md:20]:
```suggestion
A member of [Roslyn-IDE](https://github.com/orgs/dotnet/teams/roslyn-ide) will need to decide to "champion" that feature proposal and add it to the "On Deck" tab in the [IDE Design Review](ttps://github.com/dotnet/roslyn/projects/40) project. Behind the scenes the IDE Team tries to meet once a week to discuss these issues.
```

**Review by JoeRobich — APPROVED** [review]
Thanks for putting this together!

File: docs/contributing/ide_design_process.md [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+
+## Discussions / Issues
+
+The goal is that folks can create and discuss feature suggestions without needing to have everything fully figured out. Feel free to file a new feature suggestion and members of the Roslyn-IDE team will add the "[Feature Request](https://github.com/dotnet/roslyn/issues?=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22Feature+Request%22)" label. However, these feature suggestions won't be discussed until we have a proposal written for them.
+
+## Proposals
+
+### Adding a proposal
+
+For a feature proposal to be reviewed a design proposal needs be appended to the github issue. It should have the following information:
```
Comment (Youssef1313) [inline docs/contributing/ide_design_process.md:11]:
```suggestion
For a feature proposal to be reviewed, a design proposal needs be appended to the GitHub issue. It should have the following information:
```

File: docs/contributing/ide_design_process.md [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+
+- **Summary**: Brief summary of what this proposal is about.
+- **Background and Motivation**: What is the problem we are solving and in what context did you encounter it?
+- **Proposed Feature**: How the proposed solution solves the problem?
+- **Alternative Designs**: Were there other options you considered? How does this compare to features in other editors?
+
+### Getting the issue championed by the IDE team
+
+A member of [Roslyn-IDE](https://github.com/orgs/dotnet/teams/roslyn-ide) will need to decide to "champion" that feature proposal and add it to the "On Deck" tab in the [IDE Design Review](ttps://github.com/dotnet/roslyn/projects/40) project. Behind the scenes the IDE Team tries to meet once a week to discuss these issues.
```
Comment (Youssef1313) [inline docs/contributing/ide_design_process.md:20]:
```suggestion
A member of [Roslyn-IDE](https://github.com/orgs/dotnet/teams/roslyn-ide) will need to decide to "champion" that feature proposal and add it to the "On Deck" tab in the [IDE Design Review](https://github.com/dotnet/roslyn/projects/40) project. Behind the scenes the IDE Team tries to meet once a week to discuss these issues.
```

File: docs/contributing/ide_design_process.md [hunk 12]
```diff
@@ -0,0 +1,20 @@
+# IDE Team Design Process
+
+## Discussions / Issues
+
+The goal is that folks can create and discuss feature suggestions without needing to have everything fully figured out. Feel free to file a new feature suggestion and members of the Roslyn-IDE team will add the "[Feature Request](https://github.com/dotnet/roslyn/issues?=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22Feature+Request%22)" label. However, these feature suggestions won't be discussed until we have a proposal written for them.
```
Comment (333fred) [inline docs/contributing/ide_design_process.md:5]: `folks` feels a bit like slang here, maybe contributors would be a better term?

Comment (333fred) [inline docs/contributing/ide_design_process.md:5]: You say `a proposal written` here, but then refer to multiple different types of proposals below (both feature and design). Which did you mean here? Consider clarifying the difference between these types of proposals.

File: docs/contributing/ide_design_process.md [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+For a feature proposal to be reviewed, a design proposal needs be appended to the GitHub issue. It should have the following information:
+
+- **Summary**: Brief summary of what this proposal is about.
+- **Background and Motivation**: What is the problem we are solving and in what context did you encounter it?
+- **Proposed Feature**: How the proposed solution solves the problem?
+- **Alternative Designs**: Were there other options you considered? How does this compare to features in other editors?
+
+### Getting the issue championed by the IDE team
+
+A member of [Roslyn-IDE](https://github.com/orgs/dotnet/teams/roslyn-ide) will need to decide to "champion" that feature proposal and add it to the "On Deck" tab in the [IDE Design Review](https://github.com/dotnet/roslyn/projects/40) project. Behind the scenes the IDE Team tries to meet once a week to discuss these issues.
```
Comment (333fred) [inline docs/contributing/ide_design_process.md:20]: I wouldn't use `tries` here. I would just state that we meet once a week, and use `generally` as a qualifier if you think one is necessary. There are exceptions, of course, but that is our general rule.

File: .github/ISSUE_TEMPLATE/feature-suggestion.md [hunk 14]
```diff
@@ -0,0 +1,36 @@
+---
+name: Feature suggestion
```
Comment (davidwengier) [inline .github/ISSUE_TEMPLATE/feature-suggestion.md:2]: Should this say "IDE Feature suggestion", with a note about putting language feature suggestions in csharplang?

**Review by davidwengier — APPROVED** [review]

**Pull request merged**